tt_metal/impl/device/device_pool.hpp

```diff
@@ -30,6 +30,8 @@ class DevicePool {
         size_t trace_region_size = DEFAULT_TRACE_REGION_SIZE,
         const std::vector<uint32_t> &l1_bank_remap = {}) noexcept {
         log_debug(tt::LogMetal, "DevicePool initialize");
+        tt::tt_metal::dispatch_core_manager::initialize(Device::max_num_hw_cqs);
```

Comment:
If we always initialize with `max_num_hw_cqs` can't the dispatch core manager just pull that value in rather than passing it in here?

Reply:
Static linking dependency doesn't like that :(

DevicePool --> Device --> num_hw_cqs
                   \__> dispatch_core_manager

Reply:
I think I would have to define max_num_hw_cqs in dispatch_core_manager and pull that into Device

Reply:
Hmm yea that seems like a good place. I'll make the move.

Reply:
I ended up putting the constant in `dispatch_core_manager`. Didn't want `dispatch_core_manager.hpp` to include `command_queue_interface.hpp`, which includes the `dispatch_core_manager.hpp` 🤡